core/modules/image/image.module

```diff
@@ -1349,3 +1359,173 @@ function image_filter_keyword($value, $current_pixels, $new_pixels) {
+  // Get information about which fields are on this type of entity.
+  list($id, $vid, $bundle) = entity_extract_ids($entity_type, $entity);
+  $bundle_field_info = field_info_instances($entity_type, $bundle);
+
+  // Loop through each of the fields and find those that are image fields.
+  foreach ($bundle_field_info as $field_name => $instance_info) {
+    $field_info = field_info_field($field_name);
+    if ($field_info['type'] === 'image') {
+      // The langcode should normally be LANGUAGE_NONE for an image field, but
+      // check the langcode property just to be certain.
+      $langcode = LANGUAGE_NONE;
+      if (isset($entity->langcode) && isset($entity->{$field_name}[$entity->langcode])) {
+        $langcode = $entity->langcode;
+      }
+      if (!isset($entity->{$field_name}[$langcode])) {
+        continue;
+      }
+
+      // Update the field values (by reference) for SVG images.
+      $field_values = &$entity->{$field_name}[$langcode];
+      foreach ($field_values as $delta => $field_value) {
+        if (!isset($field_value['fid'])) {
+          continue;
+        }
+        $file = file_load($field_value['fid']);
+        if ($file && image_is_svg($file->uri)) {
+          // Set width/height as a safety measure. Savable NULL is better than
+          // the not savable empty string.
+          $field_values[$delta]['width'] = NULL;
+          $field_values[$delta]['height'] = NULL;
+          if ($svg_dimensions = image_get_svg_dimensions($file->uri)) {
+            $field_values[$delta]['width'] = $svg_dimensions['width'];
+            $field_values[$delta]['height'] = $svg_dimensions['height'];
+          }
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_file_url_alter().
+ *
+ * Prevent image_style_url() from using a different URL for SVG images. Because
+ * image styles cannot operate on SVGs, set the URL back to the original image.
+ * This affects theme_image_style(), which normally would point to a new image.
+ */
+function image_file_url_alter(&$url) {
+  if (!empty($url)) {
+    // If this is an SVG image in the styles directory...
+    if (strpos($url, '/styles/') && strpos($url, '.svg')) {
+      // Remove the styles directory portions and use the original image URL.
+      $url = preg_replace('!/styles/.*?/.*?/!', '/', $url);
+    }
+  }
+}
+
+/**
+ * Get an SVG image's defined dimensions.
+ *
+ * @param string $uri
+ *   The URI or path to an SVG image.
+ *
+ * @return array|FALSE
+ *   An array containing the keys "width" and "height" as integer values. If the
+ *   image is an SVG but no set dimensions are available, these keys will have
+ *   NULL values. If the image is not an SVG, FALSE will be return.
+ */
+function image_get_svg_dimensions($uri) {
+  // Safety check.
+  if (!image_is_svg($uri)) {
+    return FALSE;
+  }
+
+  // Return cached dimensions if already retrieved once this request.
+  $cached_images = &drupal_static(__FUNCTION__, array());
```

Comment:
```suggestion
  $cached_images = &backdrop_static(__FUNCTION__, array());
```

In https://github.com/backdrop/backdrop-issues/issues/6262#issuecomment-1760503696 I proposed adding `drupal` as a forbidden word in cspell, to catch these as "spelling errors" in our tests 😉